bazel/foreign_cc/BUILD

```diff
@@ -603,3 +603,15 @@ envoy_cc_library(
         "//conditions:default": [],
     }),
 )
+
+envoy_cmake(
+    name = "pugixml",
+    cache_entries = {
+        "CMAKE_BUILD_TYPE": "Release",
```

Comment:
im wondering if this builds anything unwanted - often these cmake builds build everything by default, not just the lib/etc that we need

Reply:
Looks like a better option than the prior option - thanks for researching this

Reply:
Let me look into more about how we can avoid unwanted stuff getting included.